web3/_utils/async_transactions.py

```diff
@@ -43,25 +42,28 @@
 )
 
 if TYPE_CHECKING:
-    from web3 import Web3  # noqa: F401
     from web3.eth import AsyncEth  # noqa: F401
+    from web3.main import (  # noqa: F401
```

Comment:
Here it's `from web3.main import`, but we also use `from web3 import` all over the place and I can't find a pattern. Can we just use one or the other?

Reply:
Yeah. I guess `from web3.main` might be "best" since the `web3/__init__.py` imports it from `web3.main` so it's a bit of a hand-off. I don't think it matters too much. There may be some things that are defined in `web3.main` that we don't expose via `web3/__init__.py` at some point too. For example, I'm not sure that it makes much sense to expose `BaseWeb3` there since that's not a complete class that is ready for use.